src/index.ts

```diff
@@ -58,20 +59,34 @@ async function main() {
         workflow_id,
         branch,
       });
+      console.log(`Found ${data.total_count} runs total.`);
 
-      const branchWorkflows = data.workflow_runs.filter(run => run.head_branch === branch);
-      console.log(`Found ${branchWorkflows.length} runs for workflow ${workflow_id} on branch ${branch}`);
-      console.log(branchWorkflows.map(run => `- ${run.html_url}`).join('\n'));
+      let cancel_before;
+      if (all_but_latest) {
+        cancel_before = new Date(data.workflow_runs.reduce((a, b) => Math.max(a.created_at, b.created_at)));
+      } else {
+        cancel_before = new Date(current_run.created_at);
+      }
 
-      const runningWorkflows = branchWorkflows.filter(run =>
-        (ignore_sha || run.head_sha !== headSha) &&
-        run.status !== 'completed' &&
-        new Date(run.created_at) < new Date(current_run.created_at)
+      const runningWorkflows = data.workflow_runs.filter(
+        run => run.head_branch === branch && (ignore_sha || run.head_sha !== headSha) && run.status !== 'completed' &&
+            run != current_run &&
+            new Date(run.created_at) < cancel_before
       );
-      console.log(`with ${runningWorkflows.length} runs to cancel.`);
-
-      for (const {id, head_sha, status, html_url} of runningWorkflows) {
-        console.log('Canceling run: ', {id, head_sha, status, html_url});
+      console.log(`Found ${runningWorkflows.length} runs to cancel.`);
+      for (const {id, head_sha, status} of runningWorkflows) {
+        console.log('Cancelling another run: ', {id, head_sha, status});
+        const res = await octokit.actions.cancelWorkflowRun({
+          owner,
+          repo,
+          run_id: id
+        });
+        console.log(`Cancel run ${id} responded with status ${res.status}`);
+      }
```

Comment:
This PR would be easier to review I think if there was one very focused commit separate from the others that extracted this cancel logic to separate method - then it would not be duplicated below when it is time to cancel the current workflow. Bonus points for ingesting the idea from #59 if the extracted method was not actually a single cancel but took an array of IDs / pushed the Promise result from each cancel on an array / did the Promise.all thing to alleviate some of the "github responds 202 but doesn't actually cancel right away..." issue https://github.com/styfle/cancel-workflow-action/pull/59/files#diff-3d2b59189eeedc2d428ddd632e97658fe310f587f7cb63b01f9b98ffc11c0197R5903-R5914